core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java

```diff
@@ -195,11 +195,23 @@ public boolean isContainerBuild() {
     }
 
     /**
-     * The docker image to use to do the image build
+     * The docker image to use to do the image build. It can be one of `graalvm`, `mandrel`, or the full image path, e.g.
+     * {@code quay.io/quarkus/ubi-quarkus-mandrel:21.3-java17}.
      */
     @ConfigItem(defaultValue = "${platform.quarkus.native.builder-image}")
     public String builderImage;
 
+    public String getEffectiveBuilderImage() {
+        final String builderImageName = this.builderImage.toUpperCase();
+        if (builderImageName.equals(BuilderImageProvider.GRAALVM.name())) {
+            return "quay.io/quarkus/ubi-quarkus-native-image:21.3-java11";
+        } else if (builderImageName.equals(BuilderImageProvider.MANDREL.name())) {
+            return "quay.io/quarkus/ubi-quarkus-mandrel:21.3-java11";
```

Comment:
I think it would be better to have these things into constants given we will have to change them regularly.

Reply:
FWIW, at some point, I would like to have all these properties coming from the build-parent, injected into filtered resources and loaded dynamically in the code.

That would simplify a lot the updates (it's a global problem, we have the same issue with the Dev Services images).

Reply:
@geoand @stuartwdouglas @famod do you think ^ make sense? It's out of the scope of the PR but it's a good example of this problem.

Reply:
In general, I'm all for _not_ "hiding" versions in code, be it inline or constants. Even some of explicit descriptors (e.g. codestarts IIRC) should be filtered by Maven instead.

The only downside I see with build-parent is that you end up building the entire project, although only a (tiny) fraction of modules is affected.

Reply:
I think adding it to build-parent mostly makes sense, although I do wonder if this is something that should be provided by the platform.